### ISSUE TITLE
v5 upgrade gives new error `token required because branch is protected`

### ISSUE THREAD

**Issue by jwillemsen** [codecov/codecov-action#1653]
See https://github.com/RemedyIT/taox11/pull/410 for a PR to upgrade from v4 to v5, I have a token in the secrets but with v5 it now fails with

debug - 2024-11-14 20:28:08,417 -- Commit creating result --- {"result": "RequestResult(error=RequestError(code='HTTP Error 400', params={}, description='{\"message\":\"Token required because branch is protected\"}\\n'), warnings=[], status_code=400, text='{\"message\":\"Token required because branch is protected\"}\\n')"}
error - 2024-11-14 20:28:08,417 -- Commit creating failed: {"message":"Token required because branch is protected"}

The full log

https://github.com/RemedyIT/taox11/actions/runs/11844438622/job/33007661643?pr=410

**Comment by thomasrockhu-codecov** [comment]
@jwillemsen this should be remedied in the new `5.0.1` release, let me know if that's not the case

**Comment by jwillemsen** [comment]
Upload now seems to work, now waiting on the coverage data itself

**Comment by thomasrockhu-codecov** [comment]
@jwillemsen looking at your PR on Codecov, I think I found another bug, working on a fix now

**Comment by thomasrockhu-codecov** [comment]
@jwillemsen  ok, I pushed `5.0.2` which should hopefully fix this issue

**Comment by jwillemsen** [comment]
Restarted the checks

**Comment by jwillemsen** [comment]
The github action job succeeded but still no new coverage data on codecov for the new pr

**Comment by Synchro** [comment]
5.0.2 is still failing for me [here](https://github.com/PHPMailer/PHPMailer/actions/runs/11887178853/job/33132337806?pr=3113)

**Comment by thomasrockhu-codecov** [comment]
@jwillemsen @Synchro would you be able to try `5.0.3`?

**Comment by jwillemsen** [comment]
Restarted checks

**Comment by asaf-kali** [comment]
Still happening to me with `v5.0.3`

**Comment by Synchro** [comment]
I'm not sure how to make it use 5.0.3 – I don't see a version identifier in the output [here](https://github.com/PHPMailer/PHPMailer/actions/runs/11887178853/job/33226876205). Also I can't accept the PR behind this because of [this issue](https://github.com/PHPMailer/PHPMailer/pull/3113#issuecomment-2482913961).

**Comment by asaf-kali** [comment]
@Synchro you can explicitly put `uses: codecov/codecov-action@v5.0.3` in the Gitlab workflow file.
You can see this is indeed the version used:
![Image](https://github.com/user-attachments/assets/e8f95b6b-60ad-4159-b6f6-e1f167004724)

**Comment by thomasrockhu-codecov** [comment]
@Synchro what @asaf-kali said is correct. You can also pin it to `codecov/codecov-action@v5` which will take the latest. `5.0.4` will actually display a log that shows which version you are running

**Comment by Synchro** [comment]
Thanks, I updated it to 5.0.3 and it worked, and have now bumped to 5.0.4.

**Comment by thazhemadam** [comment]
@thomasrockhu-codecov I'm running into the same issue with v5.0.4 as well. You can find the build logs [here](https://github.com/SciML/ModelingToolkit.jl/actions/runs/11927924068/job/33244079441?pr=3224#step:8:27).

**Comment by joostlek** [comment]
We seem to experience the same at home-assistant/core where every PR created from a fork just does not work. An example can be found here <https://github.com/home-assistant/core/actions/runs/11932624704/job/33258193466>

**Comment by thomasrockhu-codecov** [comment]
@joostlek are you able to try the newest version?

**Comment by thomasrockhu-codecov** [comment]
@jwillemsen @Synchro @asaf-kali same is true, do you mind trying the newest version and see if it's a problem still?

**Comment by jwillemsen** [comment]
For https://github.com/jwillemsen/daikin_onecta/pull/350, when I use the link which is in the log to the codecov upload I see the results, but still no comment on the pr and when I go to the project on codecov for the PR also no results are shown

**Comment by asaf-kali** [comment]
Still same for me with `Run codecov/codecov-action@v5.0.5`:
```
error - 2024-11-20 18:55:03,461 -- Commit creating failed: {"message":"Token required because branch is protected"}
...
error - 2024-11-20 18:55:04,503 -- Report creating failed: {"message":"Token required because branch is protected"}
...
error - 2024-11-20 18:55:05,949 -- Upload failed: {"message":"Token required because branch is protected"}
```

**Comment by thomasrockhu-codecov** [comment]
@asaf-kali for you specifically, I think you need to double-check in your repository secrets that `CODECOV_TOKEN` is not empty. That along with the newest `v5` release should fix your issue.

**Comment by thomasrockhu-codecov** [comment]
@joostlek - can you try with the latest v5 release?

@jwillemsen - still digging into this

**Comment by asaf-kali** [comment]
Can't belive I didn't verify that before 🤦🏼 
Thanks @thomasrockhu-codecov, it's working now

**Comment by thomasrockhu-codecov** [comment]
@jwillemsen I think this is the same issue as posted in https://github.com/codecov/codecov-action/issues/1662, so I'll reply to you about it there.

@joostlek just wanted to see if you had any time to test it out

**Comment by joostlek** [comment]
I did not have time this evening, but i did see a PR where someone bumped to a newer version, let me check the CI runs

**Comment by joostlek** [comment]
It seems greener than before, but I have to double check when I'm behind a PC. I'll let you know tomorrow

**Comment by thomasrockhu-codecov** [comment]
@njzjz can you try `5.0.7`? That should have been the latest

**Comment by joostlek** [comment]
It seems to be working indeed

**Comment by jwillemsen** [comment]
Works again for me, thanks for all work @thomasrockhu-codecov

**Comment by larseggert** [comment]
Still happening for us: https://github.com/mozilla/neqo/actions/runs/12348542993/job/34457586080?pr=2285#step:9:297

**Comment by thomasrockhu-codecov** [comment]
@larseggert dependabot needs access to the secret, you'll need to add it in GitHub under dependabot secrets
